code/modules/mob/living/carbon/species.dm

```diff
@@ -250,23 +253,20 @@ var/global/list/whitelisted_species = list("Human")
 				H.reagents.add_reagent("plasma", Clamp(ratio, MIN_PLASMA_DAMAGE, MAX_PLASMA_DAMAGE))
 			H.toxins_alert = max(H.toxins_alert, 1)
 	else if(O2_pp > vox_oxygen_max && name == "Vox") //Oxygen is toxic to vox.
-		var/ratio = (breath.oxygen/vox_oxygen_max) * 1000
+		var/ratio = (breath.get_moles_by_id(OXYGEN)/vox_oxygen_max) * 1000
 		H.adjustToxLoss(Clamp(ratio, MIN_PLASMA_DAMAGE, MAX_PLASMA_DAMAGE))
 		H.toxins_alert = max(H.toxins_alert, 1)
 	else
 		H.toxins_alert = 0
 
-	if(breath.trace_gases.len)	// If there's some other shit in the air lets deal with it here.
-		for(var/datum/gas/sleeping_agent/SA in breath.trace_gases)
-			var/SA_pp = (SA.moles/breath.total_moles())*breath_pressure
-			if(SA_pp > SA_para_min) // Enough to make us paralysed for a bit
-				H.Paralyse(3) // 3 gives them one second to wake up and run away a bit!
-				if(SA_pp > SA_sleep_min) // Enough to make us sleep as well
-					H.sleeping = min(H.sleeping+2, 10)
-			else if(SA_pp > 0.15)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
-				if(prob(20))
-					spawn(0) H.emote(pick("giggle", "laugh"))
-			SA.moles = 0
+	if(N2O_pp > N2O_para_min) // Enough to make us paralysed for a bit
+		H.Paralyse(3) // 3 gives them one second to wake up and run away a bit!
+		if(N2O_pp > N2O_sleep_min) // Enough to make us sleep as well
+			H.sleeping = min(H.sleeping+2, 10)
+	else if(N2O_pp > N2O_emote_min)	// There is sleeping gas in their lungs, but only a little, so give them a bit of a warning
+		if(prob(20))
+			spawn(0) H.emote(pick("giggle", "laugh"))
```

Comment:
>spawn(0)

:(
also all this code is copy-pasted from above, someone needs to fix this eventually